### PR TITLE
Refactor config

### PR DESCRIPTION
**Pull request by ldx**
This will:
* Update the config to provision milpa workers with criproxy + containerd.
* Update to Terraform 0.12 syntax.
* Use Kubernetes 1.14.

### CODE REVIEW

File: terraform/providers.tf [hunk 1]
```diff
@@ -3,18 +3,20 @@
...
 }
 
 # Using these data sources allows the configuration to be
 # generic for any region.
-data "aws_region" "current" {}
+data "aws_region" "current" {
+}
 
-data "aws_availability_zones" "available" {}
+data "aws_availability_zones" "available" {
```
Comment (justnoise) [inline terraform/providers.tf:21]: We currently can't launch t3 instances in AZ `use1-az3` any interest in blacklisting that AZ?

File: terraform/providers.tf [hunk 2]
```diff
@@ -3,18 +3,20 @@
...
+  # See https://www.terraform.io/docs/providers/aws/ for more information.
+  #  region = "us-east-1"
+  #  access_key = ""
+  #  secret_key = ""
 }
 
 # Using these data sources allows the configuration to be
 # generic for any region.
-data "aws_region" "current" {}
+data "aws_region" "current" {
```
Comment (justnoise) [inline terraform/providers.tf:18]: We currently only run in us-east-1.  Should we limit ourselves to that region?

**Review by justnoise — APPROVED** [review]
This looks great! Nice update.  Just 2 questions/comments about choosing region and AZ for our nodes.

File: terraform/providers.tf (hunk 1 above)
Comment (ldx) [reply terraform/providers.tf:21]: Yeah, I'll add a blacklist

File: terraform/providers.tf (hunk 2 above)
Comment (ldx) [reply terraform/providers.tf:18]: I'm going to add a comment about this

**Pull request merged**